### DISCUSSION TITLE
How to authenticate?

### DISCUSSION THREAD

**Discussion by mlbonniec** [nuxt/nuxt#17576] [Questions] [answered]
Hi ^^

As `nuxt/auth` is currently not available for `nuxt` v3, can someone tell me how to authenticate a user?

I saw on the roadmap that the authentication module should be released on 2022 Q2.

I tried to implements my own middleware, without any success.

I can't wait for the release, so how to do without this module?

Thanks!

**Comment by TheAlexLichter** [comment, marked as answer]
There two ways you described are the only possible ways IMO:

1. You can wait until the `auth` module is ready
2. You can roll your own authentication logic

Because it highly depends on your way of authenticating (oAuth2, stateless JWT, etc. etc.), there is no easy and "general" solution (except the `auth` module at some point) I'm afraid.

**Reply by mlbonniec** [reply]
Thanks for your response 😉

The way I planned to use is a JWT. Users will fill theirs email/password and get a JWT. It's one of the most simple authentication method.

As you're working on nuxt, do you've more informations about the release date of the `auth` module?

**Reply by TheAlexLichter** [reply]
> The way I planned to use is a JWT

Then, rolling your own authentication layer might be the way to go ☺️ 

> do you've more informations about the release date of the auth module?

I don't have more info regarding the release date I'm afraid.

**Reply by andyjamesn** [reply]
You can always use the nuxt module alternatives which make this possible now.

It includes an auth module that works pretty much identical to how it did in Nuxt 2. It uses axios and pinia store integration to make the logged in user info available on the client once logged in.

Note: look at the playground folder in each repo to get an idea how to set this up.

https://github.com/Teranode/nuxt-module-alternatives

[Auth module](https://github.com/Teranode/nuxt-module-alternatives/tree/master/%40nuxtjs-alt/auth)
